client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkUtils.java

```diff
@@ -220,4 +243,32 @@ public static <K, C> CelebornShuffleReader<K, C> createColumnarShuffleReader(
             conf,
             metrics);
   }
+
+  // Added in SPARK-32920, for Spark 3.2 and above
+  private static final DynMethods.UnboundMethod UnregisterAllMapAndMergeOutput_METHOD =
+      DynMethods.builder("unregisterAllMapAndMergeOutput")
+          .impl(MapOutputTrackerMaster.class, Integer.TYPE)
+          .orNoop()
+          .build();
+
+  // for spark 3.1 and 2.4, see detail in SPARK-32920
+  private static final DynMethods.UnboundMethod UnregisterAllMapOutput_METHOD =
+      DynMethods.builder("unregisterAllMapOutput")
+          .impl(MapOutputTrackerMaster.class, Integer.TYPE)
+          .orNoop()
+          .build();
+
+  public static void unregisterAllMapOutput(
+      MapOutputTrackerMaster mapOutputTracker, int shuffleId) {
+    if (!UnregisterAllMapAndMergeOutput_METHOD.isNoop()) {
+      UnregisterAllMapAndMergeOutput_METHOD.bind(mapOutputTracker).invoke(shuffleId);
+      return;
+    }
+    if (!UnregisterAllMapOutput_METHOD.isNoop()) {
+      UnregisterAllMapOutput_METHOD.bind(mapOutputTracker).invoke(shuffleId);
+      return;
+    }
```

Comment:
Thanks for phrasing it this way, and trying both methods !
Internally, we have backported these changes - and so our internal 3.1 actually has `unregisterAllMapAndMergeOutput` :-)